Runner/Runner.py

```diff
@@ -0,0 +1,155 @@
+_STORAGE_ACCOUNT_KEY = os.environ['PS_STORAGE_ACCOUNT_KEY']
+_SERVICE_PRINCIPAL_CREDENTIALS_CLIENT_ID = os.environ['PS_SERVICE_PRINCIPAL_CREDENTIALS_CLIENT_ID']
+_SERVICE_PRINCIPAL_CREDENTIALS_SECRET = os.environ['PS_SERVICE_PRINCIPAL_CREDENTIALS_SECRET']
+_SERVICE_PRINCIPAL_CREDENTIALS_TENANT = os.environ['PS_SERVICE_PRINCIPAL_CREDENTIALS_TENANT']
+_SERVICE_PRINCIPAL_CREDENTIALS_RESOUCE = os.environ['PS_SERVICE_PRINCIPAL_CREDENTIALS_RESOUCE']
+
+_timeout = 20
+_job_managers = []
+
+
+if __name__ == '__main__':
+    start_time = datetime.datetime.now().replace(microsecond=0)
+    print('Sample start: {}'.format(start_time))
+    print()
+    print("_BATCH_ACCOUNT_NAME", _BATCH_ACCOUNT_NAME)
+    print("_BATCH_ACCOUNT_KEY", _BATCH_ACCOUNT_KEY)
+    print("_BATCH_ACCOUNT_URL", _BATCH_ACCOUNT_URL)
+    print("_BATCH_ACCOUNT_SUB", _BATCH_ACCOUNT_SUB)
+    print("_STORAGE_ACCOUNT_NAME", _STORAGE_ACCOUNT_NAME)
+    print("_STORAGE_ACCOUNT_KEY", _STORAGE_ACCOUNT_KEY)
+    print("_SERVICE_PRINCIPAL_CREDENTIALS_CLIENT_ID",
+          _SERVICE_PRINCIPAL_CREDENTIALS_CLIENT_ID)
+    print("_SERVICE_PRINCIPAL_CREDENTIALS_SECRET",
+          _SERVICE_PRINCIPAL_CREDENTIALS_SECRET)
+    print("_SERVICE_PRINCIPAL_CREDENTIALS_TENANT",
+          _SERVICE_PRINCIPAL_CREDENTIALS_TENANT)
+    print("_SERVICE_PRINCIPAL_CREDENTIALS_RESOUCE",
+          _SERVICE_PRINCIPAL_CREDENTIALS_RESOUCE)
+    print()
+
+    # Create the blob client, for use in obtaining references to
+    # blob storage containers and uploading files to containers.
+    blob_client = azureblob.BlockBlobService(
+        account_name=_STORAGE_ACCOUNT_NAME,
+        account_key=_STORAGE_ACCOUNT_KEY)
+
+    credentials = ServicePrincipalCredentials(
+        client_id=_SERVICE_PRINCIPAL_CREDENTIALS_CLIENT_ID,
+        secret=_SERVICE_PRINCIPAL_CREDENTIALS_SECRET,
+        tenant=_SERVICE_PRINCIPAL_CREDENTIALS_TENANT,
+        resource=_SERVICE_PRINCIPAL_CREDENTIALS_RESOUCE)
+
+    batch_client = batch.BatchExtensionsClient(
+        credentials=credentials,
+        batch_account=_BATCH_ACCOUNT_NAME,
+        base_url=_BATCH_ACCOUNT_URL,
+        subscription_id=_BATCH_ACCOUNT_SUB)
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "TestConfig",
+        help="A manifest file that contains a list of all the jobs and pools you want to create.")
+    args = parser.parse_args()
+
+    # "Tests/TestConfiguration.json"
+    TestConfigurationFile = args.TestConfig
+
+    # Clean up any storage container that is older than a 7 days old.
+    try:
+        Utils.cleanup_old_resources(blob_client)
+    except Exception as e:
+        raise e
+
+    try:
+        with open(TestConfigurationFile) as f:
+            template = json.load(f)
+
+        for i in range(0, len(template["tests"])):
+            test = template["tests"][i]
+
+            applicationLicenses = None
+            try:
+                applicationLicenses = test["applicationLicense"]
```

Comment:
In the test config file 'applicationLicenses' is optional inside the Json. If it's not set the value will be a none and it will throw a KeyError. If that happens we just catch the error and move on. If we get another error we should throw it.  

I don't see a nicer way of handling this

Reply:
I think you can also do that https://stackoverflow.com/questions/25833613/python-safe-method-to-get-value-of-nested-dictionary but I'm not sure its better

Reply:
It doesn't look any better.